server/src/main/java/io/crate/analyze/where/DocKeys.java

```diff
@@ -62,7 +60,7 @@ private DocKey(int pos) {
 
         public String getId(TransactionContext txnCtx, Functions functions, Row params, SubQueryResults subQueryResults) {
             return idFunction.apply(
-                Lists.transform(
+                Lists2.map(
```

Comment:
Maybe for cases like this where the list is only consumed once it would make sense to add a `mapLazy` that is based on something like this:

```java
static class LazyMapList<I, O> extends AbstractList<O> {

    private final List<I> list;
    private final Function<? super I, ? extends O> mapper;

    public LazyMapList(List<I> list, Function<? super I, ? extends O> mapper) {
        this.list = list;
        this.mapper = mapper;
    }

    @Override
    public O get(int index) {
        return mapper.apply(list.get(index));
    }

    @Override
    public int size() {
        return list.size();
    }
}
```

Reply:
pushed the fixup. i took over this code and added the precondition to use random access lists.